wrappers/src/marshalling.cpp

```diff
@@ -128,3 +118,30 @@ realm::binding::Utf16StringAccessor::Utf16StringAccessor(const uint16_t* csbuffe
         m_size = out_begin - m_data.get();
     }
 }
+
+realm_value_t to_capi(Obj obj, SharedRealm realm)
+{
+    auto table_key = obj.get_table()->get_key();
+    auto schema = realm->schema().find(table_key);
+    if (schema == realm->schema().end())
+    {
+        // These shenanigans are only necessary because realm->schema() doesn't automatically update.
+        // TODO: remove this code when https://github.com/realm/realm-core/issues/4584 is resolved
```

Comment:
I know this is copied code, but 4584 is resolved now - perhaps we can look into removing the shenanigans 😛